### PR TITLE
[pbc] Correct non Windows build

### PR DESCRIPTION
**Pull request by decent-dcore**
* 'mpir' build dependency is Windows specific
* cleaned up deprecated stuff

### CODE REVIEW

**Comment by PhoebeHui** [comment]
@decent-dcore, thanks for the PR!

Could you include below information?
1. What issue does this PR fix? Please open an issue so others may replicate this problem.
2. Please update the baseline CI file under line https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt#L1269

File: ports/pbc/portfile.cmake [hunk 1]
```diff
@@ -1,14 +1,6 @@
 include(vcpkg_common_functions)
```
Comment (PhoebeHui) [inline ports/pbc/portfile.cmake]: Could you please remove 'include(vcpkg_common_functions)', this is deprecated.

Comment (decent-dcore) [reply ports/pbc/portfile.cmake]: ok, removed

**Comment by PhoebeHui** [comment]
@decent-dcore, it looks the CI failed on uwp is expected. Could you update the baseline CI file https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt?

pbc:arm-uwp=fail
pbc:x64-uwp=fail

**Comment by decent-dcore** [comment]
> @decent-dcore, it looks the CI failed on uwp is expected. Could you update the baseline CI file https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt?
> 
> pbc:arm-uwp=fail
> pbc:x64-uwp=fail

Done, UWP is not supported.

**Comment by PhoebeHui** [comment]
@decent-dcore, an issue required for this changes, could you file an issue?

**Review by PhoebeHui — APPROVED** [review]
The changes looks good for me.

**Comment by decent-dcore** [comment]
@PhoebeHui Why an issue is required? There are 1000+ unresolved ones, I do not think that one more makes some difference.

**Comment by PhoebeHui** [comment]
@decent-dcore, we required an issue since it's used for tracking the situation that others may encounter. eg, 'mpir' build dependency is Windows specific, however it should be port bug issue since it didn't rely on it on windows before, other users may want to know more background about it, so an issue for it may be more clear.

**Comment by decent-dcore** [comment]
@PhoebeHui Ok, here's  the issue #9348

**Comment by PhoebeHui** [comment]
Thanks!

**Review by dan-shaw — APPROVED** [review]

**Comment by dan-shaw** [comment]
LGTM, will merge after MacOS CI pipeline is back up

File: ports/pbc/portfile.cmake [hunk 2]
```diff
@@ -78,8 +68,9 @@ if(VCPKG_CMAKE_SYSTEM_NAME)
     )
 
     file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include ${CURRENT_PACKAGES_DIR}/debug/share ${CURRENT_PACKAGES_DIR}/share/info)
-    configure_file(${SOURCE_PATH}/COPYING ${CURRENT_PACKAGES_DIR}/share/pbc/copyright COPYONLY)
+    file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
 else()
+    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```
Comment (dan-shaw) [inline ports/pbc/portfile.cmake]: Why are we adding a static library check?

**Comment by PhoebeHui** [comment]
@decent-dcore , could you please resolve the conflicting files?
It seems that the static library check was added by the author when this port supported in vcpkg, commit is [77163c57](https://github.com/microsoft/vcpkg/commit/77163c57f0d31e8d5a6afd94c5134349662b86ad#diff-36bdcbf690c97841c3184f1a74ef6b25R3).

**Comment by rfric** [comment]
@PhoebeHui conflict resolved

File: ports/pbc/portfile.cmake (hunk 2 above)
Comment (rfric) [reply ports/pbc/portfile.cmake]: PBC does not export symbols so dynamic linkage has no sense

**Review by rfric — APPROVED** [review]

**Comment by PhoebeHui** [comment]
@rfric , thanks for your updates!

The failures on OSX are not related to this changes. It should not block to merge this PR.

**Pull request merged**